Classes/Task/ReIndexTaskAdditionalFieldProvider.php

```diff
@@ -187,15 +190,16 @@ public function saveAdditionalFields(
         array $submittedData,
         AbstractTask $task
     ) {
-        $this->isTaskInstanceofReIndexTask($task);
+        if ($this->isTaskInstanceofReIndexTask($task)) {
```

Comment:
I would propose to keep the usages of isTaskInstanceofIndexQueueWorkerTask as it was because otherwise we just introduce another level maybe it is an option to rename isTaskInstanceofIndexQueueWorkerTask in checkIsTaskInstanceofIndexQueueWorkerTask ?